### PR TITLE
Feature/slippage

### PR DESCRIPTION
**Pull request by Chepelau**

### CODE REVIEW

**Comment by Chepelau** [comment]
I have two questions: 
    1) Now there is an option for disabling the slippage check. Should it stay there?
    2) Should we disable check for "favourable" slippage? ie. lower price in Long, higher price in Short

**Comment by tensojka** [comment]
I don't get why do we need 3 params – should_check_slippage, max_price_diff, desired_price ?

Can't there be just desired_premia? And this is max in case of long and min in case of short. This is the same as Uniswap.

**Comment by tensojka** [comment]
Oh and did we think about deadlines?

**Comment by tensojka** [comment]
re 2) @MarekHauzr , we both don't see a reason to

**Comment by Chepelau** [comment]
Yes, if we use slippage only for one side then we could just use one param, given we don't leave an option for disabling the slippage altogether.

**Comment by MarekHauzr** [comment]
Hi,
   1) the disable slippage can be done maxing the limit for long and 0 for short... no need for should_check_slippage
   2) I can imagine cases where you don't want to have "better" price (between sending trade and trade happening the market conditions might change)... but don't add it... we will consider it as "you just got a better deal"

**Comment by MarekHauzr** [comment]
LGTM

File: contracts/amm.cairo [hunk 1]
```diff
@@ -200,7 +210,8 @@ func close_position{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_chec
     option_size : Int, // in base token
     quote_token_address: Address,
     base_token_address: Address,
-    lptoken_address: Address
+    lptoken_address: Address,
+    limit_desired_price: Math64x61_, // Should be total premia including fees - w.r.t. to opposite side
```
Comment (tensojka) [inline contracts/amm.cairo:214]: This is called a price but it's not per option, I'd call this limit_premia

**Review by tensojka — CHANGES_REQUESTED** [review]

File: contracts/amm.cairo (hunk 1 above)
Comment (Chepelau) [reply contracts/amm.cairo]: done

**Review by MarekHauzr — APPROVED** [review]

**Pull request closed without merging**